modules/tribes/tests/client/components/TribesPage.component.test.js

```diff
@@ -0,0 +1,177 @@
+/* tests to write
+
+when clicking join and not logged in, redirect to tribe page
+when clicking join and logged in, send a request to join to API
+when clicking join and logged in, update the number of tribe members
+when clicking leave, send a request to leave to API
+when clicking leave and logged in, update the number of tribe members
+
+
+*/
+
+import React from 'react';
+import * as rtl from '@testing-library/react';
```

Comment:
I guess you imported it like that to distinguish the variant that takes the container as the first argument?

I quite like [`within`](https://testing-library.com/docs/dom-testing-library/api-helpers#within-and-getqueriesforelement-apis) as another approach:

```js
import { within } from '@testing-library/react';

// ... later

const button = within(tribe).getByText('Join');
// ... or ...
const { getByText } = within(tribe)
const button = getByText('Join');
```

Reads a bit clearer to me, when I first saw `rtl` in the code I was a bit confused, and had to check the imports to find out (it has the same initialization as "right to left").

Reply:
I love the `within`. Replaced in 48216ee1f3691affb82eca49b6aaca06ed1271be.